include/boost/histogram/display.hpp

```diff
@@ -0,0 +1,212 @@
+
+const unsigned int histogram_width = 60; // 60 characters
+const float max_bin_coefficient = 0.95;  // 95% of histogram_width
+
+struct extract {
+    std::vector<std::string> upper_bounds_;
+    std::vector<std::string> lower_bounds_;
+    std::vector<int> values_;
+    unsigned int size() const {return values_.size();}
+};
+
+struct visualization_data {
+  std::vector<std::string> str_values_;
+  std::vector<int> scale_factors_;
+
+  size_t lower_bounds_width_ = 0; 
+  size_t upper_bounds_width_ = 0;
+  size_t str_values_width_ = 0;
+  size_t external_line_shift_ = 0; 
+  visualization_data(const std::vector<std::string>& str_values, 
+                     const std::vector<int>& scale_factors,
+                     const size_t& lower_bounds_width,
+                     const size_t& upper_bounds_width, 
+                     const size_t& str_values_width,
+                     const size_t& external_line_shift)
+                     :
+                     str_values_{str_values}, 
+                     scale_factors_{scale_factors},
+                     lower_bounds_width_{lower_bounds_width},
+                     upper_bounds_width_{upper_bounds_width},
+                     str_values_width_{str_values_width},
+                     external_line_shift_{external_line_shift}
+                     {}
+};
+
+template <class histogram>
+extract extract_data(const histogram& h) {
+  std::string lower, upper;
+
+  auto data = indexed(h, coverage::all);
+
+  extract ex;  
+  for (auto x : data) {
+    lower = str( boost::format("%.1f") % x.bin().lower() );
+    upper = str( boost::format("%.1f") % x.bin().upper() );
+    ex.lower_bounds_.push_back(lower);
+    ex.upper_bounds_.push_back(upper);
+    ex.values_.push_back(*x);
+  }
+  return ex;    
+}
+
+std::string get_single_label(const extract& data,
+                             const unsigned int index, 
+                             const unsigned int column_width1,
+                             const unsigned int column_width2) {
+  std::string label = "";
+  char parenthesis = ' ';
+  std::string lower = data.lower_bounds_.at(index);
+  std::string upper = data.upper_bounds_.at(index);
+  if (index == data.size() - 1)
+    parenthesis = ']';
+  else
+    parenthesis = ')';
+
+  label = '[' 
+        + str(boost::format("%s") % boost::io::group(std::setw(column_width1), lower))
+        + ", "
+        + str(boost::format("%s") % boost::io::group(std::setw(column_width2), upper))
+        + parenthesis;
+  return label;
+}
+
+std::string get_single_str_value(const std::vector<std::string>& str_values,
+                                 const unsigned int index,
+                                 const unsigned int column_width) {
+  std::string str_value = str_values.at(index);
+  str_value =
+      str(boost::format("%-s") % boost::io::group(std::setw(column_width), str_value));
+  return str_value;
+}
+
+std::vector<int> calculate_scale_factors(const std::vector<int>& values) {
```

Comment:
You can use `std::max_element` directly on the histogram.

Reply:
Thank you, I missed it.